### ISSUE TITLE
Unable to generate documents due to Handlebars error "Access has been denied to resolve the property..."

### ISSUE THREAD

**Issue by nichanank** [OpenZeppelin/solidity-docgen#132]
Installed solidity-docgen version 0.3.13
Upon doing `npx solidity-docgen` I get the following error message:
```
Handlebars: Access has been denied to resolve the property "anchor" because it is not an "own property" of its parent.
You can add a runtime option to disable the check or this warning:
See https://handlebarsjs.com/api-reference/runtime-options.html#options-to-control-prototype-access for details
Handlebars: Access has been denied to resolve the property "name" because it is not an "own property" of its parent.
You can add a runtime option to disable the check or this warning:
See https://handlebarsjs.com/api-reference/runtime-options.html#options-to-control-prototype-access for details
Handlebars: Access has been denied to resolve the property "natspec" because it is not an "own property" of its parent.
You can add a runtime option to disable the check or this warning:
See https://handlebarsjs.com/api-reference/runtime-options.html#options-to-control-prototype-access for details
Handlebars: Access has been denied to resolve the property "modifiers" because it is not an "own property" of its parent.
You can add a runtime option to disable the check or this warning:
See https://handlebarsjs.com/api-reference/runtime-options.html#options-to-control-prototype-access for details
Handlebars: Access has been denied to resolve the property "functions" because it is not an "own property" of its parent.
You can add a runtime option to disable the check or this warning:
See https://handlebarsjs.com/api-reference/runtime-options.html#options-to-control-prototype-access for details
Handlebars: Access has been denied to resolve the property "events" because it is not an "own property" of its parent.
You can add a runtime option to disable the check or this warning:
See https://handlebarsjs.com/api-reference/runtime-options.html#options-to-control-prototype-access for details
Handlebars: Access has been denied to resolve the property "ownModifiers" because it is not an "own property" of its parent.
You can add a runtime option to disable the check or this warning:
See https://handlebarsjs.com/api-reference/runtime-options.html#options-to-control-prototype-access for details
Handlebars: Access has been denied to resolve the property "ownFunctions" because it is not an "own property" of its parent.
You can add a runtime option to disable the check or this warning:
See https://handlebarsjs.com/api-reference/runtime-options.html#options-to-control-prototype-access for details
Handlebars: Access has been denied to resolve the property "ownEvents" because it is not an "own property" of its parent.
You can add a runtime option to disable the check or this warning:
See https://handlebarsjs.com/api-reference/runtime-options.html#options-to-control-prototype-access for details
...
```

The `docs` directory does get created however it doesn't document any of the functions, the contents of `index.md` that is produced is
```
[]: #
[]: #
[]: #
[]: #
[]: #
[]: #
...
```

Node v11.1.0
npm v6.4.1

**Comment by 7-of-9** [comment]
Also seeing this. 

@nichanank - did you manage to resolve?

**Comment by nknapp** [comment]
The easiest way yo solve this problem would be

* return a wrapped version of the compiled template in "handlebars.ts", that
* passes the runtime options, "allowProtoPropertiesByDefault" and "allowProtoMethodsByDefault" to the template function.

**please keep in mind that it is then possible to write templates that crash the process that runs handlebars.** I don't know if this is a relevant security issue for this project

**Comment by 7-of-9** [comment]
Hi @nknapp - thanks for the input. Is this something that can be done by a user of solidity-docgen, or if I understand rightly, this suggested fix involves code changes to solidity-docgen?

Obviously, it's completely blocking.

**Comment by nknapp** [comment]
No, as a user the only thing you can currently do is to pin the handlebars version down to 4.5.3

This is possible with yarn's resolutions feature. I'm not sure about npm.

**Comment by jjanczur** [comment]
I've got the same issue.

I've tested it with:
node v8.16.2 (and v10)
npm 6.4.1

I've downgraded handlebars but it didn't help.

Of course I've tried `solidity-docgen -i ./contracts -o ./docs`

**Comment by nichanank** [comment]
> Also seeing this.
> 
> Node: v10.16.2
> npm: 6.9.0
> 
> @nichanank - did you manage to resolve?

Extremely hacky, but my temporary workaround was to modify the code in my local `node_modules` with suggested the changes in #134. This successfully populated my `docs/index.md`  with some issues of weird formatting and broken links.

**Comment by frangio** [comment]
Fixed in #134. Thank you for reporting.
